provider/k8s/client.go

```diff
@@ -225,34 +223,35 @@ func (c *clientImpl) watch(url string, stopCh <-chan bool) (chan interface{}, ch
 	// get version
 	body, err := c.do(c.request(url))
 	if err != nil {
-		return watchCh, errCh, fmt.Errorf("failed to create request: GET %q : %v", url, err)
+		return watchCh, errCh, fmt.Errorf("failed to do version request: GET %q : %v", url, err)
 	}
 
 	var generic GenericObject
 	if err := json.Unmarshal(body, &generic); err != nil {
-		return watchCh, errCh, fmt.Errorf("failed to create request: GET %q : %v", url, err)
+		return watchCh, errCh, fmt.Errorf("failed to decode version %v", err)
 	}
 	resourceVersion := generic.ResourceVersion
 
 	url = url + "?watch&resourceVersion=" + resourceVersion
 	// Make request to Kubernetes API
 	request := c.request(url)
-	request.Transport.Dial = func(network, addr string) (net.Conn, error) {
-		conn, err := net.Dial(network, addr)
-		if err != nil {
-			return nil, err
-		}
-		// No timeout for long-polling request
-		conn.SetDeadline(time.Now())
-		return conn, nil
-	}
-	req, err := request.TLSClientConfig(c.tls).MakeRequest()
+	// request.Transport.Dial = func(network, addr string) (net.Conn, error) {
```

Comment:
@emilevauge why not removing it all together instead of leaving it commented out ? :angel: